codec/codec-fury/src/main/java/com/alipay/sofa/rpc/codec/fury/FuryHelper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.codec.fury;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import com.alipay.sofa.rpc.common.utils.ClassTypeUtils;
+import com.alipay.sofa.rpc.config.ConfigUniqueNameGenerator;
+import com.alipay.sofa.rpc.core.exception.SofaRpcRuntimeException;
+import com.alipay.sofa.rpc.log.LogCodes;
+
+/**
+ * @author lipan
+ */
+public class FuryHelper {
+    private final ConcurrentMap<String, Class<?>[]> requestClassCache  = new ConcurrentHashMap<>();
+    private final ConcurrentMap<String, Class<?>>   responseClassCache = new ConcurrentHashMap<>();
+
+    public Class<?>[] getReqClass(String service, String methodName) {
+        String key = buildMethodKey(service, methodName);
+        Class<?>[] reqClass = requestClassCache.get(key);
+        if (reqClass == null) {
+            String interfaceClass = ConfigUniqueNameGenerator.getInterfaceName(service);
+            Class<?> clazz = ClassTypeUtils.getClass(interfaceClass);
+            loadClassToCache(key, clazz, methodName);
+        } else {
+            // Check if the class loader has changed due to hot update
+            String interfaceClass = ConfigUniqueNameGenerator.getInterfaceName(service);
+            ClassLoader currentClassLoader = getClassLoader(interfaceClass);
+            if (!currentClassLoader.equals(reqClass[0].getClassLoader())) {
+                Class<?> clazz = ClassTypeUtils.getClass(interfaceClass);
+                loadClassToCache(key, clazz, methodName);
+            }
+        }
+        return requestClassCache.get(key);
+    }
+
+    public Class<?> getRespClass(String service, String methodName) {
```

Comment:
`getReqClass/getRespClass` code duplication can be merged into a separate method, you can pass map to that method:

```java
    public Class<?>[] getReqClass(String service, String methodName, ConcurrentMap<String, Class<?>[]> cache) {
        String key = buildMethodKey(service, methodName);
        Class<?>[] reqClass = requestClassCache.get(key);
        if (reqClass == null) {
            String interfaceClass = ConfigUniqueNameGenerator.getInterfaceName(service);
            Class<?> clazz = ClassTypeUtils.getClass(interfaceClass);
            loadClassToCache(key, clazz, methodName);
        } else {
            // Check if the class loader has changed due to hot update
            String interfaceClass = ConfigUniqueNameGenerator.getInterfaceName(service);
            ClassLoader currentClassLoader = getClassLoader(interfaceClass);
            if (!currentClassLoader.equals(reqClass[0].getClassLoader())) {
                Class<?> clazz = ClassTypeUtils.getClass(interfaceClass);
                loadClassToCache(key, clazz, methodName);
            }
        }
        return cache.get(key);
    }
```

Reply:
good idea!

Reply:
good idea!